### DISCUSSION TITLE
coupled simulation for CFD and XFEM

### DISCUSSION THREAD

**Discussion by waynezw0618** [idaholab/moose#22874] [Q&A Getting Started] [answered]
Hi everyone
I am new to MOOSE. I used to things like OpenFOAM to CFD and FEniCS et al for FEM. So that FSI problems can be coupled with the two.  and I am wondering whether I can use MOOSE to perform simulation of crack of a floating body with squeezing and crash

Best

**Comment by GiudGiud** [comment, marked as answer]
Hello

What do you mean by squeezing? and crash?
You can likely code something in MOOSE, but there is no existing models for floating bodies for example

Guillaume
